lib/IRGen/Linking.cpp

```diff
@@ -1037,7 +1037,11 @@ llvm::Type *LinkEntity::getDefaultDeclarationType(IRGenModule &IGM) const {
   case Kind::NoncanonicalSpecializedGenericTypeMetadataCacheVariable:
     return IGM.TypeMetadataPtrTy;
   case Kind::TypeMetadataDemanglingCacheVariable:
-    return llvm::StructType::get(IGM.Int32Ty, IGM.Int32Ty);
+    if (IGM.getModule()->getDataLayout().isBigEndian()) {
```

Comment:
Is this a big-endian thing or an AVR thing?

Reply:
It does seem to match the code in MetadataRequest.cpp:

```
    if (IGM.getModule()->getDataLayout().isBigEndian()) {
      structBuilder.addInt32(-mangledStringSize);
      structBuilder.addRelativeAddress(mangledString);
    } else {
      structBuilder.addRelativeAddress(mangledString);
      structBuilder.addInt32(-mangledStringSize);
    }
```

Reply:
Ah! Thanks, John